src/parser/monk/mistweaver/modules/talents/Upwelling.tsx

```diff
@@ -130,7 +134,7 @@ class Upwelling extends Analyzer {
     }
 
     if(this.combatants.players[targetID].hasBuff(SPELLS.ESSENCE_FONT_BUFF.id, event.timestamp, 0, 0)) {//do they have the hot
-      if(this.hotMap.get(targetID).fullCount || event.timestamp - BASE_HOT_TIME > this.hotMap.get(targetID).applicationTime){
+      if(this.hotMap.get(targetID)?.fullCount || event.timestamp - BASE_HOT_TIME > this.hotMap.get(targetID)!.applicationTime){
```

Comment:
```suggestion
    const hot = this.hotMap.get(targetID);
    if(hot === undefined) {
      return;
    }

    if(this.combatants.players[targetID].hasBuff(SPELLS.ESSENCE_FONT_BUFF.id, event.timestamp, 0, 0)) {//do they have the hot
      if(hot.fullCount || event.timestamp - BASE_HOT_TIME > hot.applicationTime){
```

something like this could save you from the `?` and `!`

Reply:
Agree